src/util/xeval/evaluator.rs

```diff
@@ -303,6 +305,31 @@ impl Evaluator {
         let (left, right) = try!(self.eval_two_children(ctx, expr));
         eval_arith(left, right, f)
     }
+
+    fn eval_case_when(&mut self, ctx: &EvalContext, expr: &Expr) -> Result<Datum> {
+        for chunk in expr.get_children().chunks(2) {
+            let res = try!(self.eval(ctx, &chunk[0]));
+            if chunk.len() == 1 {
```

Comment:
seem to be different with https://github.com/pingcap/tidb/pull/1856/files#diff-9c6a66f0086282f3806a4c7ed4eecf05R44?